### PR TITLE
BCDA-3582 - Enable BFD v2 patient endpoint

### PR DESCRIPTION
**Pull request by mtrang1263**
### Fixes [BCDA-3582](https://jira.cms.gov/browse/BCDA-3582)

BFD just released the v2 patient resource. This PR enables our v2 endpoints to leverage it. We had anticipated that Coverage would also be available. It is not. We will create a follow up PR to enable it.

This is a follow up on the [PR](https://github.com/CMSgov/bcda-app/pull/570) which set up the framework to allow us to send v1 and v2 BFD requests.

### Change Details

1. Change over the request path for BFD v2 endpoints to actually use v2.  It was set to v1 because Patient EP was not available.
2. Update our smoke test client to accept API version.
3. Update smoke test client to emit response code and body in case of an error. This helps better troubleshoot failures.

### Security Implications

- [ ] new software dependencies
- [ ] security controls or supporting software altered
- [x] new data stored or transmitted

This will be the first opportunity for users to retrieve FHIR r4 data.

- [ ] security checklist is completed for this change
- [ ] requires more information or team discussion to evaluate security implications
- [ ] no PHI/PII is affected by this change

### Acceptance Validation
1. Added two additional smoke tests that request our v2 endpoints.
2. Validated that v2 requests for patient resource requests the data using the BFD v2 endpoint.
3. Deployed this branch to dev and ran smoke tests that requested v2 resources. [Smoke test](https://bcda-ci.adhocteam.us/job/BCDA%20-%20Test%20-%20Integration%20Smoke/2526/).

### CODE REVIEW

File: bcda/api/v2/api.go
Comment (msnook) [inline bcda/api/v2/api.go:21]: What does this mean for production?  It is my understanding that BFD only deployed the `v2/Patient` resource in `prod-sbx`, but not `prod`.

Comment (mtrang1263) [reply bcda/api/v2/api.go:21]: We'll leave this PR open until the v2 patient is deployed to prod

Comment (msnook) [reply bcda/api/v2/api.go:21]: Should we change it to `Draft PR`?

Comment (mtrang1263) [reply bcda/api/v2/api.go:21]: I think it's OK to have it reviewed and approved. We just won't merge it into master until we get the greenlight from BFD.

Comment (mtrang1263) [reply bcda/api/v2/api.go:21]: > What does this mean for production? It is my understanding that BFD only deployed the `v2/Patient` resource in `prod-sbx`, but not `prod`.

Followed up with the BFD team. There's no immediate plan to roll out v2 endpoints to production. For `prod`, we will not enable v2 endpoints (keep the `VERSION_2_ENDPOINT_ACTIVE` set to `false`), but enable the other env (dev, test, sbx).

**Review by msnook — APPROVED** [review]
Looks good!  Thanks for adding the negative tests to the Postman suite.

**Review by PatKelsh — APPROVED** [review]

**Pull request merged**
